lib/matplotlib/axes/_base.py

```diff
@@ -2501,34 +2501,50 @@ def _update_title_position(self, renderer):
                 x, y = title.get_position()
                 if not np.isclose(y, 1.0):
                     self._autotitlepos = False
-                    _log.debug('not adjusting title pos because title was'
+                    _log.debug('not adjusting title pos because a title was'
                              ' already placed manually: %f', y)
                     return
             self._autotitlepos = True
 
+        ymax = -10
         for title in titles:
             x, y0 = title.get_position()
-            y = 1.0
+            y = 1
+            # need to start again in case of window resizing
+            title.set_position((x, 1.0))
             # need to check all our twins too...
             axs = self._twinned_axes.get_siblings(self)
 
+            top = 0  # the top of all the axes twinned with this axes...
             for ax in axs:
                 try:
                     if (ax.xaxis.get_label_position() == 'top'
                             or ax.xaxis.get_ticks_position() == 'top'):
                         bb = ax.xaxis.get_tightbbox(renderer)
-                        top = bb.ymax
-                        # we don't need to pad because the padding is already
-                        # in __init__: titleOffsetTrans
-                        yn = self.transAxes.inverted().transform((0., top))[1]
-                        y = max(y, yn)
+                    else:
+                        bb = ax.get_window_extent(renderer)
+                    top = max((top, bb.ymax))
```

Comment:
`max(top, bb.ymax)` would work as well, i.e. multiple arguments instead of a tuple. It's even slightly more clean.